### PR TITLE
libCloudAccumulate

### PR DESCRIPTION
**Pull request by gtinchev**
- Implemented changes from the other cloud accumulate classes in the lib. 
- Added function to convert b/w pronto pc and pcl pc.

fyi @simalpha

### CODE REVIEW

**Comment by simalpha** [comment]
I pulled both of the branches (gt-lib-cl-acc from gtinchev/pronto and gt-lib-cl-acc from gtinchev/registration) and reviewed the code. These updates work for me. I am happy to merge [Pull #11](https://github.com/openhumanoids/registration/pull/11) once these modifications get integrated in Pronto.

@mauricefallon: Could you merge this PR?

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.hpp [hunk 1]
```diff
@@ -63,6 +70,7 @@ class CloudAccumulate{
     pronto::PointCloud* combined_cloud_;
     
     bool finished_;
+    long long int timeFinished_;
```
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.hpp]: for clarity and consistency we use int64_t for times and would call the variable utimeFinished_

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.hpp [hunk 2]
```diff
@@ -27,13 +27,19 @@ struct CloudAccumulateConfig
...
+
     ~CloudAccumulate(){
-    }    
+      delete laser_projector_;
+      delete projected_laser_scan_;
+    }
     
     int getCounter(){ return counter_; }
     bool getFinished(){ return finished_; }
+    long getFinishedTime() { return timeFinished_; }
```
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.hpp]: change timeFinished_ to utimeFinished_

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp [hunk 3]
```diff
@@ -190,11 +217,8 @@ void CloudAccumulate::processVelodyne(const  bot_core::pointcloud2_t* msg){
   combined_cloud_->points.insert(combined_cloud_->points.end(), scan_local->points.begin(), scan_local->points.end());
   
   counter_++;  
-  if (counter_ > ca_cfg_.batch_size){
-    std::cout << "Finished Collecting: " << msg->utime << "\n";
+  if (counter_ >= ca_cfg_.batch_size){
+    timeFinished_ = msg->utime;
```
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: utimeFinished

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp [hunk 4]
```diff
@@ -120,32 +141,38 @@ pronto::PointCloud*  CloudAccumulate::convertPlanarScanToCloud(std::shared_ptr<b
...
+
   // Accumulate
   combined_cloud_->points.insert(combined_cloud_->points.end(), scan_local->points.begin(), scan_local->points.end());
   
-  counter_++;  
-  if (counter_ > ca_cfg_.batch_size){
-    std::cout << "Finished Collecting: " << this_msg->utime << "\n";
+  counter_++;
+  if (counter_ >= ca_cfg_.batch_size){
+    timeFinished_ = this_msg->utime;
```
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: utime...

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp [hunk 5]
```diff
@@ -120,32 +141,38 @@ pronto::PointCloud*  CloudAccumulate::convertPlanarScanToCloud(std::shared_ptr<b
   return scan_local;
 }
 
-
 void CloudAccumulate::processLidar(const  bot_core::planar_lidar_t* msg){
   
-  if (!frame_check_tools_.isLocalToScanValid(botframes_)){
+  // TODO: This has to be re-implemented in botframes, as it checks for 3 recent transforms:
```
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: clearly frame_check_tools_.isLocalToScanValid() is not general.

you dont need to implement it now, but could you:
- add a boolean command line argument to enable/disable check? (to retain previous behaviour)
- make an issue on rpg-nav?

Comment (simalpha) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: @mauricefallon I updated the code according to your suggestions.

Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: Isn't LocalToScanValid still disabled?

Otherwise I'm ready to merge

Comment (simalpha) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: Yes it is. Sorry I thought you did not want me to do it now.
Okay I will add a command line argument now.

**Comment by simalpha** [comment]
@mauricefallon Are these the changes you were expecting?

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp (hunk 5 above)
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: thanks!

**Pull request merged**